src/internal/handlers.ts

```diff
@@ -88,6 +90,14 @@ export function registerBeforeUnloadHandler(handler: (readyToUnload: () => void)
   handler && sendMessageToParent('registerHandler', ['beforeUnload']);
 }
 
+export function registerUserSettingsChangeHandler(
+  settingKeys: UserSettingKeys[],
+  handler: (settingKey: UserSettingKeys, value: any) => void,
```

Comment:
We should use registerHandler, which ca be extended to take params/args as part of register handler.